### PR TITLE
feat: create CRUD operations for hotels

### PR DESCRIPTION
**Pull request by dushimsam**
#### What does this PR do?
 
Contains the CRUD operations for Hotel's model.

#### Description of  the task to be completed?

The task is comprised of CREATE, GET ALL, GET & UPDATE APIs for hotel's model

#### How should this be manually tested?

 1. Run the project - npm run dev
 2. Open the swagger editor on [http](http://localhost:[PORT]/api-docs)
 3. Run the APIs on the Hotel's section
 
#### Any background context you want to provide?

...

#### What are the relevant pivotal tracker stories?

#184779354

#### Screenshots (if appropriate)
image
image
image
image
image

...

### CODE REVIEW

**Review by honorine22 — APPROVED** [review]
Great work!

File: src/database/migrations/20230327163941-create-hotel.js [hunk 1]
```diff
@@ -0,0 +1,60 @@
...
+      },
+      cell:{
+        type: Sequelize.STRING
+      },
+      village:{
+        type: Sequelize.STRING
+      },
+      coordinates: {
+        type: Sequelize.STRING
+      },
```
Comment (Nkbtemmy) [inline src/database/migrations/20230327163941-create-hotel.js]: what if you have latitude and longitude separately !!

Comment (Nkbtemmy) [reply src/database/migrations/20230327163941-create-hotel.js]: when yo do that make sure they have correct datatypes

File: src/database/models/hotel.js [hunk 2]
```diff
@@ -0,0 +1,79 @@
...
+    allowNull: true,
+  },
+  village:{
+    type: DataTypes.STRING,
+    allowNull: true,
+  },
+  coordinates: {
+    type: DataTypes.STRING,
+    allowNull: true,
+  },
```
Comment (Nkbtemmy) [inline src/database/models/hotel.js]: Same as here

File: src/database/seeders/20230327164033-sample-hotels.js [hunk 3]
```diff
@@ -0,0 +1,50 @@
...
+          name: "The Grand Hotel",
+          email: "reservations@thegrandhotel.com",
+          address: "123 Main St, City, State",
+          website: "https://www.thegrandhotel.com",
+          province: "Kigali",
+          district: "Gasabo",
+          sector: "Kacyiru",
+          cell: "Kamatamu",
+          village: "Karukina",
+          coordinates: "POINT (23.456789 1.234567)",
```
Comment (Nkbtemmy) [inline src/database/seeders/20230327164033-sample-hotels.js]: ....

**Review by Nkbtemmy — CHANGES_REQUESTED** [review]

File: src/database/migrations/20230327163941-create-hotel.js (hunk 1 above)
Comment (dushimsam) [reply src/database/migrations/20230327163941-create-hotel.js]: Done

image

image

It's working in posting data via other channels such as APIs, but failing when seeding data.

image

**Review by Nkbtemmy — APPROVED** [review]

**Review by oreste-abizera — APPROVED** [review]
LGTM

**Pull request merged**